### PR TITLE
create UserInfoCard component

### PR DESCRIPTION
**Pull request by abdulrahman-2020**
- create a card using antd

- add icons using fontawesome-react library

### CODE REVIEW

File: client/src/components/profile/UserInfoCard.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { Card, Image } from 'antd';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar, faStarHalf } from '@fortawesome/free-solid-svg-icons';
+import { faFacebook, faInstagram, faWhatsapp } from '@fortawesome/free-brands-svg-icons';
+
+import profileImg from '../../assets/profileImg.png';
+import 'antd/dist/antd.css';
+import '../../App.css';
+
+function UserInfoCard() {
```
Comment (LinaYahya) [inline client/src/components/profile/UserInfoCard.tsx:11]: build your component to receive props,
and In the place you call add these data as props

File: client/src/components/profile/UserInfoCard.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    <div>
+      <Card
+        bordered
+        style={{ width: 300 }}
+      >
+        <Image width={100} src={profileImg} />
+        <FontAwesomeIcon icon={faStar} size="lg" />
+        <FontAwesomeIcon icon={faStar} size="lg" />
+        <FontAwesomeIcon icon={faStar} size="lg" />
+        <FontAwesomeIcon icon={faStarHalf} size="lg" />
```
Comment (LinaYahya) [inline client/src/components/profile/UserInfoCard.tsx:22]: check rate component at antd

File: client/src/components/profile/UserInfoCard.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        <span>
+          <FontAwesomeIcon icon={faWhatsapp} size="lg" />
+          {' '}
+          تواصل معي
+        </span>
+        <br />
+        <span>
+          <FontAwesomeIcon icon={faStar} size="lg" />
+          {' '}
+          أضف تقيماً
```
Comment (LinaYahya) [inline client/src/components/profile/UserInfoCard.tsx:54]: translate any text you have

File: client/src/Pages/Profile.tsx [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
+import UserInfoCard from '../components/profile/UserInfoCard';
 
 function Profile() {
   return (
     <div className="container">
       {useTranslation().t('profile-greeting')}
+      <UserInfoCard />
```
Comment (LinaYahya) [inline client/src/Pages/Profile.tsx:9]:
```suggestion
      <UserInfoCard userInfo={{image, name, ...etc}} />
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by AbdallahGot — APPROVED** [review]

**Review by RandSohail — APPROVED** [review]

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:1]: rename file to index.tsx

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import {
+  Card, Image, Rate,
+} from 'antd';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+import { faFacebook, faInstagram, faWhatsapp } from '@fortawesome/free-brands-svg-icons';
+
+import { useTranslation } from 'react-i18next';
+import 'antd/dist/antd.css';
+import '../../App.css';
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:12]: No need to import stylesheet here as you import it at app.tsx

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import { Link } from 'react-router-dom';
+import {
+  Card, Image, Rate,
+} from 'antd';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+import { faFacebook, faInstagram, faWhatsapp } from '@fortawesome/free-brands-svg-icons';
+
+import { useTranslation } from 'react-i18next';
+import 'antd/dist/antd.css';
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:11]: remove this line as you already import it in the App.tsx

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+          <Image width={100} src={image1} />
+        </div>
+
+        <div className="content">
+          <div className="name">
+            <h2>{name}</h2>
+            <Rate allowHalf defaultValue={rate} disabled className="rate" />
+          </div>
+          <p>{location}</p>
+          {useTranslation().t('servicesOffer')}
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:49]: invoke useTranslation out of return

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+          </p>
+          <p>
+            {description}
+          </p>
+        </div>
+        <div className="contact">
+          <p>{email}</p>
+          <p>{phone}</p>
+          <div className="social">
+            <Link to={instagram}><FontAwesomeIcon style={{ color: '#AA38A5' }} icon={faInstagram} size="2x" /></Link>
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:65]: while we're navigating outside our route use an anchor tag instead.

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    </Card>
+  );
+}
+
+// UserInfoCard.propTypes = {
+//   userInfo: PropTypes.shape({
+//     name: PropTypes.string.isRequired,
+//     city: PropTypes.string.isRequired,
+//   }).isRequired,
+// };
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:93]: remove comments

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+import { faFacebook, faInstagram, faWhatsapp } from '@fortawesome/free-brands-svg-icons';
+
+import { useTranslation } from 'react-i18next';
+import 'antd/dist/antd.css';
+import '../../App.css';
+import image1 from '../../assets/profileImg.png';
+import './userInfocard.css';
+
+interface IUserInfoCardProps {
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:16]:
```suggestion
interface UserInfoCardProps {
```

File: client/src/components/UserInfoCard/UserInfoCard.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+} from 'antd';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStar } from '@fortawesome/free-solid-svg-icons';
+import { faFacebook, faInstagram, faWhatsapp } from '@fortawesome/free-brands-svg-icons';
+
+import { useTranslation } from 'react-i18next';
+import 'antd/dist/antd.css';
+import '../../App.css';
+import image1 from '../../assets/profileImg.png';
+import './userInfocard.css';
```
Comment (LinaYahya) [inline client/src/components/UserInfoCard/UserInfoCard.tsx:14]: rename your file to style.css

File: client/package.json [hunk 13]
```diff
@@ -12,6 +17,7 @@
     "@types/react-dom": "^18.0.1",
     "antd": "^4.19.5",
     "axios": "^0.26.1",
+    "babel-plugin-macros": "^3.1.0",
```
Comment (LinaYahya) [inline client/package.json:20]: why you install it?

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by devstm — APPROVED** [review]

**Pull request merged**